services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/DescribeTableEnhancedResponse.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.model;
+
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbAsyncTable;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbTable;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableResponse;
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * Defines the elements returned by DynamoDB from a {@code DescribeTable} operation, such as
+ * {@link DynamoDbTable#describeTable()} and {@link DynamoDbAsyncTable#describeTable()}
+ */
+@SdkPublicApi
+public final class DescribeTableEnhancedResponse {
+    private final DescribeTableResponse response;
+
+    private DescribeTableEnhancedResponse(Builder builder) {
+        this.response = Validate.paramNotNull(builder.response, "response");
+    }
+
+    /**
+     * The {@link DescribeTableResponse} from the low level DynamoDB client
+     */
+    public DescribeTableResponse describeTableResponse() {
```

Comment:
Can we remove these? The other `Enhanced*Response` objects don't have it. I think ideally we don't want this on the enhanced response because we should expose the relevant stuff on the enhanced object if we need to.

Reply:
I think all fields in the `DescribeTableResponse` is relevant to a describeTable call. You mean copying all fields in `DescribeTableResponse` here?  My concern is that we will need to make code change in the future if new fields are added

Reply:
Since the enhanced client is supposed to replace the majority of the use cases for the low level client, I feel like asking the customer to work with the low level response goes against the spirit of that.

```
enhancedResponse.describeTableResponse().tableName();
```

Also looks awkward to me

Reply:
For other operations such as getItem, yes, but I'm not sure about control plane APIa. Do you have any suggestions? I'd be hesitant to manually copying all the fields. It's a lot of ops burden because they can be out-of-date soon

Reply:
Would you be okay with just keeping a reference to the low level response internally, and just exposing direct getters on the object? Best middle ground I can think of

```
public String tableName() {
    return lowLevelResponse.tableName();
}
```

Reply:
`DescribeTableResponse` only has `TableDescription` object actually. I guess we can do 
```java
public String table() {
    return lowLevelResponse.table();
}
```